### ISSUE TITLE
Typescript type definitions incorrect for Command class

### ISSUE THREAD

**Issue by abedra** [tj/commander.js#787]
When compiling a typescript project the following error is produced

```
node_modules/commander/typings/index.d.ts(26,25): error TS2689: Cannot extend an interface 'NodeJS.EventEmitter'. Did you mean 'implements'?
```

**Comment by ivankostov** [comment]
Same issue for me - commander - 2.15.1, typescript 2.8.1.
Workaround is to remove the extends statement 
before:
`
class Command extends NodeJS.EventEmitter {
`
after:
`
class Command {
`
Command does not implement any of the EventEmmiter's functions.

**Comment by avendiart** [comment]
Adding `@types/node` to `devDependencies` has fixed the issue for us.

**Comment by kushwahashiv** [comment]
I also get an error ```TypeError: cmd.command is not a function```

**Comment by arzyu** [comment]
`@types/node` should be added as a dependency in `commender.js`. As a temporary solution, I add it to my own package before this done.

**Comment by shadowspawn** [comment]
We should document that intend end-user to install @types/node.

**Comment by shadowspawn** [comment]
Added TypeScript tips in README. Hopefully that helps.
https://github.com/tj/commander.js#typescript

**Comment by 8Observer8** [comment]
> Same issue for me - commander - 2.15.1, typescript 2.8.1.
> Workaround is to remove the extends statement
> before:
> `class Command extends NodeJS.EventEmitter {`
> after:
> `class Command {`
> Command does not implement any of the EventEmmiter's functions.

Thank you! This solution works for Electron 9 and Node 12.

**Comment by shadowspawn** [comment]
For future readers still using older versions of Commander, I'll add a link to the older README instructions:
https://github.com/tj/commander.js/tree/v4.1.1#typescript

The TypeScript dependency on `NodeJS.EventEmitter` was removed in #1146 which shipped in Commander v5.0.0.
